File: solutions/lua/raindrops/2/raindrops.lua
return function(n)
    local result = ""

    if n % 3 == 0 then
        result = result .. "Pling"
    end
    if n % 5 == 0 then
        result = result .. "Plang"
    end
    if n % 7 == 0 then
        result = result .. "Plong"
    end

    if result == "" then
        return string.format(n)
    else
        return result
    end
end

File: solutions/ruby/raindrops/2/raindrops.rb
class Raindrops
  def self.convert(n)

    str = ""
    if n % 3 == 0
      str += "Pling"
    end

    if n % 5 == 0
      str += "Plang"
    end

    if n % 7 == 0
      str += "Plong"
    end

    return n.to_s if str.empty? 
    
    str
  end
end